### PR TITLE
Fix JsonWebToken.CipherText returns String.Empty for decrypted token

### LINKED ISSUE #1942: [Bug] Regression from 6.21.0 to 6.23.1
**Which version of Microsoft.IdentityModel are you using?**
Note that to get help, you need to run the latest version. 

**Where is the issue?**
 * [x ] M.IM.JsonWebTokens
 * [ ] M.IM.KeyVaultExtensions
 * [ ] M.IM.Logging
 * [ ] M.IM.ManagedKeyVaultSecurityKey
 * [ ] M.IM.Protocols
 * [ ] M.IM.Protocols.OpenIdConnect
 * [ ] M.IM.Protocols.SignedHttpRequest
 * [ ] M.IM.Protocols.WsFederation
 * [ ] M.IM.TestExtensions
 * [ ] M.IM.Tokens
 * [ ] M.IM.Tokens.Saml
 * [ ] M.IM.Validators
 * [ ] M.IM.Xml
 * [ ] S.IM.Tokens.Jwt
 * Other (please describe)

6.21.0 -> JsonWebToken.CipherText returns null for decrypted token
6.23.1 -> JsonWebToken.CipherText returns String.Empty for decrypted token.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #1942.

### CODE REVIEW

**Comment by joegoldman2** [comment]
@jennyf19 @brentschmaltz this PR is ready to be reviewed.

**Comment by keegan-caruso** [comment]
@joegoldman2 thanks for the contribution. Would love to see a simple test.

**Comment by joegoldman2** [comment]
> @joegoldman2 thanks for the contribution. Would love to see a simple test.

For sure! I tried but struggle, I'll try again and come back to you.

**Comment by joegoldman2** [comment]
> @joegoldman2 thanks for the contribution. Would love to see a simple test.

For sure! I tried but struggle. I'll try again and come back to you.

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs [hunk 1]
```diff
@@ -115,6 +115,43 @@ public void JWETouchAllProperties()
...
+            if (isEncrypted)
+            {
+                Assert.NotNull(jsonWebToken.AuthenticationTag);
+                Assert.NotNull(jsonWebToken.AuthenticationTagBytes);
+                Assert.NotNull(jsonWebToken.Ciphertext);
+                Assert.NotNull(jsonWebToken.CipherTextBytes);
+            }
+            else
+            {
+                Assert.Empty(jsonWebToken.AuthenticationTag);
```
Comment (joegoldman2) [inline test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs:148]: Should we apply the same thing for AuthenticationTag? Currently `string.Empty` is returning is the token is not encrypted.

**Comment by joegoldman2** [comment]
> @joegoldman2 thanks for the contribution. Would love to see a simple test.

done

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs [hunk 2]
```diff
@@ -180,6 +180,43 @@ public void JWETouchAllProperties()
...
```
Comment (joegoldman2) [inline test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenTests.cs:213]: Should we apply the same thing for AuthenticationTag? Currently `string.Empty` is returning if the token is not encrypted.

**Comment by joegoldman2** [comment]
@jennyf19 should we keep this PR open or the version 6x is already out of support?

**Comment by joegoldman2** [comment]
No feedback for 8 months, I believe the version 6 is no longer supported. Closing this PR.

**Pull request closed without merging**